common/metrics/src/metric_server.rs

```diff
@@ -42,25 +63,64 @@ fn serve_metrics(req: Request<Body>) -> impl Future<Item = Response<Body>, Error
     future::ok(resp)
 }
 
-pub fn start_server<T: ToSocketAddrs>(to_addr: T) {
-    let addr: SocketAddr = to_addr
+fn serve_public_metrics(
+    req: Request<Body>,
+) -> impl Future<Item = Response<Body>, Error = hyper::Error> {
+    let mut resp = Response::new(Body::empty());
+    match (req.method(), req.uri().path()) {
+        (&Method::GET, "/public_metrics") => {
+            //Create a different path to expose public metrics
+            let encoder = TextEncoder::new();
+            let buffer = encode_public_metrics(encoder);
+            *resp.body_mut() = Body::from(buffer);
+        }
+        _ => {
+            *resp.status_mut() = StatusCode::NOT_FOUND;
+        }
+    };
+
+    future::ok(resp)
+}
+
+pub fn start_server(host: String, port: u16, public: bool) {
+    let addr: SocketAddr = (host.as_str(), port)
         .to_socket_addrs()
-        .unwrap_or_else(|_| panic!("Failed to parse address"))
+        .unwrap_or_else(|_| panic!("Failed to parse {}:{} as address", host, port))
         .next()
         .unwrap();
 
-    rt::run(rt::lazy(move || {
-        match Server::try_bind(&addr) {
-            Ok(srv) => {
-                let srv = srv
-                    .serve(|| service_fn(serve_metrics))
-                    .map_err(|e| eprintln!("server error: {}", e));
-                info!("Metric server listening on http://{}", addr);
-                rt::spawn(srv);
-            }
-            Err(e) => error!("Metric server bind error: {}", e),
-        };
+    match public {
+        false => {
```

Comment:
Id recommend using an `if/else` block here instead of a match block. It ends up reducing the indent level by one and makes it more readable.